### ISSUE TITLE
lock_jars([:runtime], { :resolve => false }) is returning transitive dependencies

### ISSUE THREAD

**Issue by r6p** [mguymon/lock_jar#33]
I don't think this should be happening per:

````
LockJar.list(*args): Lists all dependencies as notations for groups from the Jarfile.lock. Depending on the type of arg, a different configuration is set.

[String] will set the Jarfile.lock path, e.g. 'Better.lock'. Default lock file is 'Jarfile.lock'.
[Array] will set the groups, e.g. ['default', 'runtime']. Defaults group is default.
[Hash] will set the options, e.g. { :local_repo => 'path' }
:local_repo [String] sets the local repo path. Defaults to ENV['M2_REPO'] or '~/.m2/repository'
:local_paths [Boolean] converts the notations to paths of jars in the local repo
:resolve [Boolean] to true will make transitive dependences resolve before returning list of jars
````

**Comment by r6p** [comment]
Could this be due to the buildr dsl using the dependency list from the local lock file?

**Comment by r6p** [comment]
Yup, it seems that once the "dependencies" are added to the respective lockfile, this DSL method always returns the transitive dependencies.

**Comment by mguymon** [comment]
Correct, by default it always reads the `Jarfile.lock`. The `resolve` parameter is only useful when inlining jar dependencies:

    LockJar.list(resolve: true) do 
         jar 'org.apache.tinkerpop:gremlin-core:3.1.0-incubating'
     end

Are you attempting to just read the jars in the Jarfile?

**Comment by r6p** [comment]
Reading Jars in the Jarfile might work. It is a bit of a hack, but I can workaround it with this snippet of code. I would prefer having a way to access only the top-level/non-transitive dependencies for packaging purposes. Any simple way to add functionality to support resolve => false for non-inlining cases?

````
  # Only return the direct/non-transitive dependencies
  lockfile            = LockJar::Domain::Lockfile.read(lockfile_name)
  direct_dependencies = lockfile.groups[group.to_s]['artifacts'].collect{ |ar| ar.keys}.flatten
  direct_dependencies.collect {|dep| dep.gsub(/^jar:/, '')}
````

**Comment by mguymon** [comment]
Sorry for the slow reply, I was under the weather last week. :pill: 

Checkout this [PR](https://github.com/mguymon/lock_jar/pull/34) that should address the issue.

**Comment by r6p** [comment]
I think it look ok, with quick pass through the one commit. Do you want me to do a more detailed review?

**Comment by mguymon** [comment]
@r6p what ever you feel comfortable with

**Comment by mguymon** [comment]
Released version 0.14.6
